HopsanGUI/Dialogs/OptimizationScriptWizard.cpp

```diff
@@ -1356,20 +1364,20 @@ QString OptimizationScriptWizard::generateFunctionCode(int i)
 
 //! @brief Checks if number of selected variables is correct. Gives error messages if they are too many or too low.
 //! @param i Selected objective function
-bool OptimizationScriptWizard::verifyNumberOfVariables(int idx, int nSelVar)
+bool OptimizationScriptWizard::verifyNumberOfVariables(int idx, int nSelVar, bool noWarning)
 {
     int nVar = mObjectiveFunctionNumberOfVariables.at(idx);
 
-    if(nSelVar > nVar)
-    {
-        //This error makes no sense, since it is showed on the last page only, so user will see it too late
-        //mpMessageHandler->addErrorMessage("Too many variables selected for this function.");
-        return false;
-    }
-    else if(nSelVar < nVar)
+    if(nSelVar != nVar)
     {
-        //This error makes no sense, since it is showed on the last page only, so user will see it too late
-        //mpMessageHandler->addErrorMessage("Too few variables selected for this function.");
+        if(!noWarning)
+        {
+            QMessageBox::warning(this, QString("Wrong number of arguments"),
+                                 QString::fromStdString("Selected function requires exactly "+
+                                 std::to_string(nVar)+
+                                 " variables, but you selected "+
+                                 std::to_string(nSelVar)+"."),QMessageBox::Ok);
```

Comment:
This would be easier to read if something like: QString("text %1 more text %2").arg(v1).arg(v2) was used instead of mixing QString and std:string

Also I think that there should be a "The" before "selected"